pkg/code-generator/collector/collector.go

```diff
@@ -181,15 +88,14 @@ func (c *collector) importsForProtoFile(absoluteRoot, protoFile string, customIm
 			return nil, err
 		}
 		dependency := filepath.Join(importPath, importedProto)
-		dependencyImports, err := c.importsForProtoFile(absoluteRoot, dependency, customImports)
+		dependencyImports, err := c.synchronizedImportsForProtoFile(absoluteRoot, dependency, customImports)
 		if err != nil {
-			return nil, errors.Wrapf(err, "getting imports for dependency")
+			return nil, eris.Wrapf(err, "getting imports for dependency")
 		}
 		importsForProto = append(importsForProto, strings.TrimSuffix(importPath, "/"))
 		importsForProto = append(importsForProto, dependencyImports...)
 	}
-
-	return importsForProto, nil
+	return stringutils.Unique(importsForProto), nil
```

Comment:
this feels like a behavior change. how did we handle duplicates before?

Reply:
We used to do this further out, in the compiler: https://github.com/solo-io/solo-kit/pull/417/files#diff-c1b7cbb4edcfe824cf0146bf0190676e1b9b2d7489ddcb828f082c45b1b5928aL103